scripts/eurostat/gdp/import_data.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Downloads and cleans GDP data from the Eurostat database.
+
+    Usage:
+
+    python3 import_data.py
+"""
+
+import pandas as pd
+import json
+from preprocess_data import preprocess_df
+
+class EurostatGDPImporter:
+    """Pulls GDP data from the Eurostat database. Specifically, it processes the
+    data in the nama_10r_3gdp folder (available from the Eurostat website, linked
+    in README).
+
+    Attributes:
+        df: DataFrame (DF) with the cleaned data.
+    """
+    DATA_LINK = "https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?file=data/nama_10r_3gdp.tsv.gz"
+    UNIT_CODES = {
+        "MIO_EUR":	         "Million euro",
+        "EUR_HAB":	         "Euro per inhabitant",
+        # "EUR_HAB_EU":	     "Euro per inhabitant in percentage of the EU average",
+        # "EUR_HAB_EU27_2020": "Euro per inhabitant in percentage of the EU27 (from 2020) average",
+        "MIO_NAC":	         "Million units of national currency",
+        "MIO_PPS":	         "Million purchasing power standards (PPS)",
+        # "MIO_PPS_EU27_2020": "Million purchasing power standards (PPS, EU27 from 2020)",
+        "PPS_HAB":	         "Purchasing power standard (PPS) per inhabitant",
+        # "PPS_EU27_2020_HAB": "Purchasing power standard (PPS, EU27 from 2020), per inhabitant",
+        # "PPS_HAB_EU":	     "Purchasing power standard (PPS) per inhabitant in percentage of the \
+        #                       EU average",
+        # "PPS_HAB_EU27_2020": "Purchasing power standard (PPS, EU27 from 2020), per inhabitant in \
+        #                       percentage of the EU27 (from 2020) average"
+    }
+    DESIRED_COLUMNS = ['geo', 'time'] + list(UNIT_CODES.keys())
+
+    # TODO (fpernice): Add a human-readable description of the codes to statVarObs
+    REGION_CODES = dict(json.loads(open('region_codes.json').read()))
+
+    def __init__(self):
+        self.raw_df = None
+        self.preprocessed_df = None
+        self.clean_df = None
+        print(self.REGION_CODES['EU27_2020'])
+
+    def download_data(self):
+        """Downloads raw data from Eurostat website and stores it in instance
+        data frame.
+        """
+        self.raw_df = pd.read_table(self.DATA_LINK)
```

Comment:
Very cool! Did not know that pandas could read in compressed files. Thanks for teaching me this! I think this could also feed back to Jin's script and remove some unzipping, etc. code.

Reply:
Yep, it's pretty nice. Note the use of `pd.read_table()` instead of e.g. `pd.read_csv()`.